opendbc/car/hyundai/interface.py

```diff
@@ -25,7 +25,17 @@ def _get_params(ret: structs.CarParams, candidate, fingerprint, car_fw, experime
     # FIXME: the Optima Hybrid 2017 uses a different SCC12 checksum
     ret.dashcamOnly = candidate in {CAR.KIA_OPTIMA_H, }
 
-    hda2 = Ecu.adas in [fw.ecu for fw in car_fw]
+    # Check if the ADAS Driving ECU is present in the firmware response.
+    # If not found, check the CAM CAN bus fingerprint for known HDA2 steering messages (0x50 or 0x110),
+    # that indicate an HDA2 car even without a direct ADAS Driving ECU response.
```

Comment:
Is the CAM CAN bus check reliable in all cases? If so, we should dispense with the FW check entirely.

Reply:
Yes, these two steering messages only exist on HDA2 cars, and we only care about if they are on bus 2 (tx).

Reply:
Yes, we should only have one source

Reply:
@sunnyhaibin setting to draft while you make that update, I'll validate when you're finished.

Reply:
@jyoung8607 Updated with 1 source only and unit test